File: apis/demo-service/src/main/java/com/example/demo_service/Customer.java
package com.example.demo_service;

import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 顧客情報
 */
@Entity
@Table(name = "CUSTOMERS")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    @Column(name = "NAME", nullable = false, length = 255)
    private String name;

    @OneToMany(mappedBy = "customers")
    @Column(nullable = true)
    private List<Sale> sales;
}

File: demo-front/src/types/shops/Customer.ts
import type { Sale } from "@/types/shops/Sale";

export interface Customer {
  id: number;
  name: string;
  sales: Sale[];
}

// interfaceだとエラー, PickとOmitはtypeでないと受け取れない
export type CustomerSelection = Pick<Customer, "name" | "id">;
